### ISSUE TITLE
Move projects between groups and subgroups

### ISSUE THREAD

**Issue by effemmeffe** [gitlabform/gitlabform#213]
Basically the title says all wht I'd like: can I move one project or even more than one projects from one group to another group?

**Comment by effemmeffe** [comment]
I agree with your proposal, but I don't have the technical ability to write the code, sorry.
I'll keep the notification active on this issue to monitor any change in the future, thanks.

**Comment by amimas** [comment]
> For moving all projects from one group to another we could do:
> 
> ```
> projects_and_groups:
>  "group_one/* -> group_two/*": 
> ```
>
> This would require group_two to already exist (creating completely new groups and projects is a different feature, in my opinion) and would break if any project would already exist in group_two.

I'm not sure how often this use case is needed. If I ever have the above use case, I'd probably just [rename the group](https://docs.gitlab.com/ee/user/group/manage.html#change-a-groups-path) `group_one` and its path to `group_two`. No need for individually moving projects. Either way, I don't see a big need for this "feature" in gitlabform.

I do see a need for moving individual projects from one group to another. Currently gitlabform config syntax is at v3. I propose a syntax like this:

```yaml
projects_and_groups:
  group_two/project_bar:
    project_settings:
      ... # regular project setting here

    project:
      trasnsfer_from: group_one/project_foo 
```

I chose to put the new config under `project` key because it's already used for other administrative task of a project (i.e. `archive: true`). We could discuss whether the new config should be `transfer_to` instead of `transfer_from`. I chose `transfer_from` because I think it makes sense for project "config" to reflect it's final location.

**Comment by amimas** [comment]
@gdubicki - I was thinking of taking a stab at this if time permits. Does the above make sense?

One thing I don't like about `transfer_from` config is that it's kinda opposite of the GitLab API and the UI. In GitLab, it's more of a transfer **to** instead of **from**.

Aside from that, implentation-wise I'm not sure how to get gitlabform to process `group_two/project_bar` when the project initially doesn't exist at that location. Can you point me to where that processing will need to be tweaked?

**Comment by amimas** [comment]
This is now available in [v3.9.0 release](https://github.com/gitlabform/gitlabform/releases/tag/v3.9.0).
